### ISSUE TITLE
Scratch space isolation between standard jobs

### ISSUE THREAD

**Issue by henricasanova** [wrench-project/wrench#46]
We need to clarify semantics for scratch space. Here is the proposal:
  - a compute service has a single scratch storage of a given size specified at construction time
  - When a standard job runs on the compute service (NOT within a pilot job), that standard job has its own temporary "directory" in scratch. As a result, the same file could be stored multiple times in scratch, each copy for a different standard job.  This requires that a scratch storage service provide a bit more functionality than a normal storage service.
  - When standard jobs run within a pilot job  then these standard jobs shared a single scratch temporary directory (and if one of them wipes out a file another needs, then too bad, that's the WMS's fault).

So, in a nutshell, we need to extend the StorageService and/or SimpleStorageService API  and implementation to include a "temp directories" abstraction, to be defined.

**Comment by henricasanova** [comment]
done
